### PR TITLE
ENH: Common DatetimeIndex generation for test instruments -- pysat 3.0.0 edition

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #359.  Twin to #365.

After #365 is merged, some other changes are needed here.  In particular, make sure that version caps do not propagate through to pysat 3.0.0.

## Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

See #365 

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,6 +18,8 @@ addons:
 before_install:
   - pip install coveralls
   - pip install future
+  - pip install 'pandas<0.25'
+  - pip install 'xarray<0.15'
```
Comment (jklenzing) [inline .travis.yml]: Make sure these are removed before merge into develop-3

**Comment by jklenzing** [comment]
NOTE: remove deprectaed functions before merge.

**Comment by jklenzing** [comment]
This should be 3.0 compliant now.

**Comment by jklenzing** [comment]
The 2.x version of this (#365) had been approved and merged.  Any comments before I merge the 3.0 version in?

File: pysat/instruments/methods/testing.py [hunk 2]
```diff
@@ -19,6 +111,9 @@ def generate_fake_data(t0, num_array, period=5820, data_range=[0.0, 24.0],
         Not used for non-cyclic functions.
         (default = 24.0)
     cyclic : bool
+        If True, assume that fake data is a cyclic function (ie, longitude,
+        slt) that will reset to data_range[0] once it reached data_range[1].
```
Comment (rstoneback) [inline pysat/instruments/methods/testing.py]: small typo: reaches

**Pull request merged**